mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -102,12 +102,11 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
 impl<E: std::error::Error + Send + Sync> PartQueueProducer<E> {
     /// Push a new [Part] onto the back of the queue
     pub fn push(&self, part: Result<Part, PrefetchReadError<E>>) {
-        let part_size = part.as_ref().ok().map(|part| part.len());
-        let send_result = self.sender.send_blocking(part);
-        if let Some(part_size) = part_size {
-            self.bytes_sent.fetch_add(part_size, Ordering::SeqCst);
+        if let Ok(ref part) = part {
```

Comment:
nit: you can also use:
```suggestion
        if let Ok(part) = &part {
```